File: src/main/java/io/cosmospf/gatling/amqp/javaapi/request/AmqpDslBuilderBase.java
package io.cosmospf.gatling.amqp.javaapi.request;

public final class AmqpDslBuilderBase {
    private final io.cosmospf.gatling.amqp.request.AmqpDslBuilderBase wrapped;

    public AmqpDslBuilderBase(io.cosmospf.gatling.amqp.request.AmqpDslBuilderBase wrapped){
        this.wrapped = wrapped;
    }

    public PublishDslBuilderExchange publish(){
        return new PublishDslBuilderExchange(wrapped.publish(io.gatling.core.Predef.configuration()));
    }
    public RequestReplyDslBuilderExchange requestReply(){
        return new RequestReplyDslBuilderExchange(wrapped.requestReply(io.gatling.core.Predef.configuration()));
    }
}

File: src/main/scala/org/galaxio/gatling/amqp/request/AmqpDslBuilderBase.scala
package org.galaxio.gatling.amqp.request

import io.gatling.core.config.GatlingConfiguration
import io.gatling.core.session._

case class AmqpDslBuilderBase(requestName: Expression[String]) {
  def publish(implicit configuration: GatlingConfiguration): PublishDslBuilderExchange           =
    PublishDslBuilderExchange(requestName, configuration)
  def requestReply(implicit configuration: GatlingConfiguration): RequestReplyDslBuilderExchange =
    RequestReplyDslBuilderExchange(requestName, configuration)
}
